### PR TITLE
Add check 102 to detect `SY-SYSID` usage

### PR DESCRIPTION
**Pull request by ConjuringCoffee**
This new check finds logic utilizing the value from `SY-SYSID`. 

I'm currently preparing for a system migration, so the name of the system will be different from before. I want to use this check to fix potential errors where logic is coupled to a specific system name.

### CODE REVIEW

File: src/checks/zcl_aoc_check_102.clas.locals_imp.abap [hunk 1]
```diff
@@ -0,0 +1,80 @@
+CLASS lcl_check_helper IMPLEMENTATION.
+  METHOD constructor.
+    mo_scan = io_scan.
+  ENDMETHOD.
+
+  METHOD determine_error_code.
+    IF is_using_only_first_letter( is_token ).
```
Comment (larshp) [inline src/checks/zcl_aoc_check_102.clas.locals_imp.abap:7]: do downport of the IFs first, then I guess the other errors will become more useful

**Comment by ConjuringCoffee** [comment]
I'm not sure how to proceed with RFC enablement: Would I just call `enable_rfc( )` in the constructor and that's it? I have no idea what the limitations are.

**Comment by larshp** [comment]
> I'm not sure how to proceed with RFC enablement: Would I just call `enable_rfc( )` in the constructor and that's it? I have no idea what the limitations are.

yes, for this scenario

RFC works well when its just the statements and tokens that are used

File: src/checks/zcl_aoc_check_102.clas.locals_def.abap [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  PUBLIC SECTION.
+    METHODS constructor
+      IMPORTING
+        io_scan TYPE REF TO zcl_aoc_scan.
+
+    METHODS determine_error_code
+      IMPORTING
+        is_token             TYPE stokesx
+        iv_index_token       TYPE syst_tabix
```
Comment (larshp) [inline src/checks/zcl_aoc_check_102.clas.locals_def.abap:26]:
```suggestion
        iv_index_token       TYPE sy-tabix
```

or alternatively

```suggestion
        iv_index_token       TYPE i
```

`syst_tabix` is not released

**Review by larshp — APPROVED** [review]
looks good to me, ship it :)

**Comment by ConjuringCoffee** [comment]
Thanks for the quick responses 🎉

**Comment by larshp** [comment]
@ConjuringCoffee you should have access to create branches in this repository now

commit early and often, open a PR as draft, to get early feedback from the linter

**Pull request merged**